src/timetable_gui.cpp

```diff
@@ -47,15 +49,35 @@ struct TimetableArrivalDeparture {
  */
 void SetTimetableParams(int param1, int param2, TimerGameTick::Ticks ticks)
 {
-	if (_settings_client.gui.timetable_in_ticks) {
-		SetDParam(param1, STR_TIMETABLE_TICKS);
-		SetDParam(param2, ticks);
-	} else {
-		SetDParam(param1, STR_TIMETABLE_DAYS);
-		SetDParam(param2, ticks / Ticks::DAY_TICKS);
+	switch (_settings_client.gui.timetable_mode) {
+		case TimetableMode::Days:
+			SetDParam(param1, STR_UNITS_DAYS);
+			SetDParam(param2, ticks / Ticks::DAY_TICKS);
+			break;
+		case TimetableMode::Seconds:
+			SetDParam(param1, STR_UNITS_SECONDS);
+			SetDParam(param2, ticks / Ticks::TICKS_PER_SECOND);
+			break;
+		case TimetableMode::Ticks:
+			SetDParam(param1, STR_UNITS_TICKS);
+			SetDParam(param2, ticks);
+			break;
+		default:
+			NOT_REACHED();
 	}
 }
 
+/**
+ * Get the correct number of ticks depending on the timetable mode.
+ * @return The number of ticks per day or second, depending on which time units are displayed.
+ */
+static inline TimerGameTick::Ticks TicksPerDayOrSecond()
+{
+	if (_settings_client.gui.timetable_mode == TimetableMode::Seconds) return Ticks::TICKS_PER_SECOND;
+
+	return Ticks::DAY_TICKS;
+}
```

Comment:
This function seems to be used for two purposes:

1) to convert from ticks to days or seconds (but not for display, as that's handled above).
2) to set the threshhold for lateness.

I think the second case should be handled in a separate function (something like `TimetableLatenessThreshold()`), and this function should be able to used to convert a value to/from days, seconds, or ticks (return 1)

If you do that then there are some special cases further on that no longer need to be special cases.